cpp/src/parquet/CMakeLists.txt

```diff
@@ -202,6 +203,19 @@ set(PARQUET_SRCS
     stream_writer.cc
     types.cc)
 
+if(CXX_SUPPORTS_AVX2)
+  # AVX2 is used as a proxy for BMI2.
```

Comment:
According to https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html , it seems that the following CPUs may have support for AVX2 but not BMI2: Intel Alder Lake, Via Eden. This is obviously an extremely minor fraction of the CPU market, but I wouldn't be surprised if some day we get some bug reports because of it (or worse, people will decide that Arrow isn't reliable because it crashes for some of their customers).

Reply:
@pitrou I added a comment below about why I think the current config should be OK (and why people should not copy the flags).  At this point, I believe the only place BMI2 is enabled is on level_conversion_bmi2.cc  which at this point should always be guarded by runtime dispatch that specifically checks for BMI2.  Thoughts?

Reply:
Ahah, sorry. Indeed, for some reason I had misread that `-mbmi2` would be applied to all source files. I agree it's ok, then.